src/figuresClasses.ts

```diff
@@ -1,19 +1,77 @@
-export interface Figure {
+type FigureColor = 'red' | 'green' | 'blue';
+type FigureShape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: FigureShape;
+  color: FigureColor;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: FigureShape = 'triangle';
 
-}
+  constructor(
+    public color: FigureColor,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0
+      || b <= 0
+      || c <= 0
+      || (a + b) <= c
+      || (c + a) <= b
+      || (c + b) <= a
+    ) {
+      throw new Error('Wrong sides!');
```

Comment:
Checklist p3 
The rest is the same

How users should understand what exactly was wrong and what data are valid?